jsmpp/src/main/java/org/jsmpp/bean/OptionalParameter.java

```diff
@@ -256,23 +256,32 @@ public String getValueAsString() {
     public static class COctetString extends OctetString {
 
         public COctetString(short tag, String value, String charsetName)
-                throws UnsupportedEncodingException {
-            super(tag, value, charsetName);
-        }
+				throws UnsupportedEncodingException {
+			super(tag, new byte[value.getBytes(charsetName).length + 1]);
+			System.arraycopy(value.getBytes(charsetName), 0, this.value, 0,
+					value.getBytes(charsetName).length);
+			this.value[value.getBytes().length] = (byte) 0x00;
```

Comment:
not passing the charsetName argument to getBytes() on that line

You call value.getBytes(charsetName) many times, can you revise the code to call it just once?